### PR TITLE
Handle file not found error on SGA

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
Hi

In this PR I am handling exception, if file gets delete on storage then I am showing 500 status code with error message  **"File not found message"**

https://github.com/mitodl/edx-sga/issues/100
CC @pdpinch @carsongee 

screen shot 2015-06-29 at 6 10 13 pm

screen shot 2015-06-29 at 6 13 26 pm

### CODE REVIEW

**Comment by pdpinch** [comment]
@amir-qayyum-khan can we catch this error and give the user a more descriptive error message? For debugging I'd actually like to know the name of the file and the bucket that wasn't found, but that may expose too much information to the end user. 

@carsongee is a 404 a more appropriate response than a 500?

**Comment by pdpinch** [comment]
Suggested user-friendly error message:

Error message for students (i.e. : "Sorry, the file you uploaded, <file name>, cannot be found. Please try uploading it again or contact course staff"

Error message for instructors (i.e. grading view): "Sorry, assignment <file name> cannot be found at <location>. Please contact  <support e-mail>"

**Comment by pdpinch** [comment]
FYI @pwilkins

**Comment by amir-qayyum-khan** [comment]
Done @pdpinch

**Comment by pdpinch** [comment]
@amir-qayyum-khan can you rebase this when you get a chance?

**Comment by amir-qayyum-khan** [comment]
sure

**Comment by amir-qayyum-khan** [comment]
Done 

cc @pdpinch

**Comment by jetej** [comment]
How might I delete a file from storage to test this?
When a student replaces an ungraded file with another, the instructor can attempt to download the first file but will actually download the second one.

**Comment by pdpinch** [comment]
I'm not sure if your test sandbox is using S3 or local storage. @carsongee should know.

**Comment by carsongee** [comment]
It is using local, `/edx/var/edxapp/uploads` is where they all are

**Comment by jetej** [comment]
As an instructor, I viewed the student's submission using the "Grade Submissions" panel.  I deleted the submitted assignments using the local path `/edx/var/edxapp/uploads`.  I returned as instructor to view the submitted assignments before refreshing the page.  It drew the IOError below.

I tried refreshing the page and the deleted submissions were still displaying in the "Grade Submissions panel."

I can attempt to view the deleted submitted file as a student but I receive an IOError as well.

Staff error:
screen shot 2015-07-23 at 11 05 07 am

Student error:
screen shot 2015-07-23 at 11 17 54 am

**Comment by jetej** [comment]
Oops, so sorry! Ignore my last comment.  It works fine.  I was a bit clumsy about testing the correct PR for this issue.

**Comment by amir-qayyum-khan** [comment]
@jetej you can delete comments, if these are irrelevant. To avoid confession.

File: edx_sga/sga.py [hunk 1]
```diff
@@ -508,20 +513,36 @@ def staff_download_annotated(self, request, suffix=''):
...
-                                 filename.encode('utf-8')))
+        try:
+            file_descriptor = default_storage.open(path)
+            app_iter = iter(partial(file_descriptor.read, BLOCK_SIZE), '')
+            return Response(
+                app_iter=app_iter,
+                content_type=mime_type,
+                content_disposition="attachment; filename=" + filename.encode('utf-8'))
+        except IOError:
+            if require_staff:
```
Comment (pdpinch) [inline edx_sga/sga.py]: Why is this conditional here? Students can download their own uploads, so there is the potential of an error for them as well.

Comment (pdpinch) [inline edx_sga/sga.py]: Oh, wait. I see the student response in L540-545

**Comment by pdpinch** [comment]
What happens if the instructor's annotated file is lost?

**Comment by amir-qayyum-khan** [comment]
@pdpinch then he gets staff related error message.

**Pull request merged**